pragma-node/src/handlers/entries/get_onchain/mod.rs

```diff
@@ -62,18 +75,25 @@ pub async fn get_onchain(
     .map_err(|db_error| db_error.to_entry_error(&pair_id))?;
 
     // TODO(akhercha): ⚠ gives different result than onchain oracle sometime
-    let last_updated_timestamp =
-        get_last_updated_timestamp(&state.onchain_pool, params.network, raw_data.pair_used)
-            .await
-            .map_err(|db_error| db_error.to_entry_error(&pair_id))?;
+    let last_updated_timestamp = get_last_updated_timestamp(
+        &state.onchain_pool,
+        params.network,
+        raw_data[0].pair_used.clone(),
+    )
+    .await
+    .map_err(|db_error| db_error.to_entry_error(&pair_id))?;
 
-    Ok(Json(adapt_entries_to_onchain_response(
-        pair_id,
-        raw_data.decimal,
-        raw_data.sources,
-        raw_data.price,
-        last_updated_timestamp,
-    )))
+    let mut api_result: Vec<GetOnchainResponse> = Vec::new();
```

Comment:
we already know the total size:
```suggestion
    let mut api_results: Vec<GetOnchainResponse> = Vec::with_capacity(raw_data.len());
```

Reply:
done